include/cuco/detail/static_set/static_set.inl

```diff
@@ -307,6 +307,80 @@ void static_set<Key, Extent, Scope, KeyEqual, ProbingScheme, Allocator, Storage>
       first, num_keys, output_begin, ref(op::find));
 }
 
+template <class Key,
+          class Extent,
+          cuda::thread_scope Scope,
+          class KeyEqual,
+          class ProbingScheme,
+          class Allocator,
+          class Storage>
+template <typename InputIt, typename OutputIt1, typename OutputIt2>
+std::pair<OutputIt1, OutputIt2>
+static_set<Key, Extent, Scope, KeyEqual, ProbingScheme, Allocator, Storage>::retrieve(
+  InputIt first,
+  InputIt last,
+  OutputIt1 output_probe,
+  OutputIt2 output_match,
+  cuda_stream_ref stream) const
+{
+  auto res = this->retrieve_async(first, last, output_probe, output_match, stream);
+  stream.synchronize();
+  return res;
+}
+
+template <class Key,
+          class Extent,
+          cuda::thread_scope Scope,
+          class KeyEqual,
+          class ProbingScheme,
+          class Allocator,
+          class Storage>
+template <typename InputIt, typename OutputIt1, typename OutputIt2>
+std::pair<OutputIt1, OutputIt2>
```

Comment:
```suggestion
cuda::std::pair<OutputIt1, OutputIt2>
```

Although I'm not sure if this is helpful or not

Reply:
This is a host API thus returning a host device pair seems an overkill.

Reply:
Right, a plain `std::pair` will work here. My point is more about design guidelines. `cuda::std::` is a superset of `std::` as it works both on host and device. My understanding is that we should use `cuda::std::` as much as possible for this reason. Although I'm not sure if there's any automatic namespace aliasing `std:: -> cuda::std::` happening or not.

Reply:
> cuda::std:: is a superset of std:: as it works both on host and device.

That's exactly the reason I tent to use, e.g.  `cuda::std::pair` for host device API, `thrust::pair` (or is there a `cuda::pair`?) for device API and `std::pair` for host only.

Reply:
Switching to `cuda::std::pair` for all APIs based on offline discussions